apis/v1alpha1/validation.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"errors"
+	"fmt"
+	"net/mail"
+	"strings"
+)
+
+// allowedResource defines the resource supported.
+var allowedResource = map[string]struct{}{
+	"organizations": {},
+	"folders":       {},
+	"projects":      {},
+}
+
+// ValidateIAMRequest checks if the ResourceMapping is valid.
+func ValidateIAMRequest(r *IAMRequest) (retErr error) {
+	for _, s := range r.ResourcePolicies {
+		// Check if resource type is valid.
+		if _, contains := allowedResource[strings.Split(s.Resource, "/")[0]]; !contains {
```

Comment:
I would suggest splitting the `strings.Split(s.Resource)` and map lookup into two lines. But I also agree with @yolocs's feedback below that using a map might be overkill for this use case.